### PR TITLE
Return only layers authorized to token provided in the URL

### PR DESCRIPTION
**Pull request by nahuelange**

### CODE REVIEW

File: geostore/tokens.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+from .models import Layer, LayerGroup
+
+
+class GroupTilesToken(TokenGenerator):
+
+    def get_groups_intersect(self, user, layergroup):
+        return Group.objects.filter(
+            Q(authorized_layers__in=Layer.objects.filter(layer_groups__in=[layergroup])) &
+            Q(pk__in=[g.pk for g in user.groups.all()])
```
Comment (submarcos) [inline geostore/tokens.py]:
```suggestion
            Q(pk__in=user.groups.all())
```

**Review by submarcos — COMMENTED** [review]
is it not possible to avoid terra_utils dependency ?

in all cases, it requires test (at least 100% diff hits)

**Comment by nahuelange** [comment]
> is it not possible to avoid terra_utils dependency ?

We added a TokenGenerator utility class in terra_utils, since it could be used in many projects. Put it in geostore seems to be inappropriate. Maybe we can create a package for this, but it seems overkill.

File: geostore/tokens.py (hunk 1 above)
Comment (nahuelange) [reply geostore/tokens.py]: fixed

**Comment by nahuelange** [comment]
token generator was splitted in a standalone python package named `terra-token-tools`

**Pull request merged**